### DISCUSSION TITLE
Airflow 1.10.1 - MySQL V8 computability

### DISCUSSION THREAD

**Discussion by avico78** [apache/airflow#32860] [Q&A]
Current airflow V 1.10.11
MySQL 5.7
Due to plans for upgrading MySQL to v8 

I have concern for computability with - following documentations
V 1.10.11 support MySQL versions: 5.6, 5.7
see: https://pypi.org/project/apache-airflow/1.10.11/#requirements

As check , 
The minimum Airflow version that support  MySQL 8  is : 1.10.14
https://pypi.org/project/apache-airflow/1.10.14/#requirements

although not mention - Is Airflow 1.10.11 compatible with MySQL 8?
If not, are there any potential risks in upgrading from Airflow **1.10.11 to 1.10.14** and then upgrading MySQL to version 8?

**Comment by potiuk** [comment]
Airflow 1.10 is not supported  already for almost 3 years - it reached End-Of-Life long time ago. If you take it literally - your setup is not supported at all for almost 3 years AT ALL. not only for any mysql version. And a lot of dependencies (like MySQL) that you want to upgrade will not work. the more you wait with the migration the more problems you will have and the less lkely you will find answer to those problems (you will not get them here for sure because - airflow 1.10 is not supported for years so no-one here will remember anything about it and they will have 0 incentive to help you. S 

So if you are worried about supportability, my advice is to upgrade to Airlow 2.6 ASAP. You are using old, out-dated software that has not been supported for years and you will not get any support (including any help if you migrate to 1.10.14. 

Airflow website has full documentation about upgrading to Airflow 2 (searh for it). Follow it.

Of course you might want to stick with Airlfow 1.10 but you are painting yourself to this corner:

![image](https://github.com/apache/airflow/assets/595491/09d937e8-eb1d-4a69-acbb-4e18a78ae0aa)

**Reply by avico78** [reply]
Thank you, @potiuk, for your response.
 I fully agree with your points. However, we are also considering replacing our Airflow solution with something else. As a result, investing too much effort into it might not be justifiable. Unfortunately, the upgrade to MySQL V8 is non-negotiable, so I'm currently seeking the most straightforward solution.

**Reply by potiuk** [reply]
Sure. There are other solutions available. If you find it easier to migrate to them and they will give you better value - it's entirely your choice. Just compare it with the latest Airflow and see which migration is cheaper for you. It's your choice to make.

**Reply by potiuk** [reply]
There is only so long you can accumulate technical debt. It'just seems that you run out of your options and you must start paying it back.

**Comment by avico78** [comment]
@potiuk, I believe the best approach would be to conduct thorough testing:

First, upgrade MySQL to V8 and check if AirFlow V 1.10.11 works with it.
If it doesn't, then proceed with upgrading AirFlow from 1.10.11 to 1.10.14 and upgrade MySQL to V8.
again , if there was plan to keep Airflow - i would go on  2.6 but this is a  temporary situation till the whole solution change to something else .

**Reply by potiuk** [reply]
Well if you know better, that's your choice. It's not likely to work and you will have problems that will be very difficult to investigate, but hey - nobody can neither force you in doing/not doing it. nor help you if you choose that path. You are entirely on your own if you do.

You've been warned.
